Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -45,6 +49,46 @@ async function run() {
         var isFolderBasedDeployment = await isInputPkgIsFolder(webDeployPkg);
         var publishingProfile = await azureRmUtil.getAzureRMWebAppPublishProfile(SPN, webAppName, resourceGroupName, deployToSlotFlag, slotName);
         tl._writeLine(tl.loc('GotconnectiondetailsforazureRMWebApp0', webAppName));
+        
+        if(xmlTransformationAndVariableSubstitution && xmlTransformation) {
```

Comment:
In order to prevent unnecessary zip-unzip when the user checks on xmlTransformationAndVariableSubstitution but doesnt check either the xdtTransformation or variableSubstitution